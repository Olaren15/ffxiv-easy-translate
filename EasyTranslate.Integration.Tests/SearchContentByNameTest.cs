using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Threading.Tasks;
using EasyTranslate.Domain.Entities;
using EasyTranslate.UseCase;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Xunit;

namespace EasyTranslate.Integration.Tests;

// Assertions are done with snapshots
[SuppressMessage("Blocker Code Smell", "S2699:Tests should include assertions")]
public class SearchContentByNameTest
{
    private readonly SearchContentByNameUseCase _searchContentByNameUseCase;

    public SearchContentByNameTest()
    {
        IServiceProvider serviceProvider = ServiceProviderBuilder.Build();
        _searchContentByNameUseCase = serviceProvider.GetService<SearchContentByNameUseCase>()!;
    }

    [Fact]
    public async Task Item_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Popoto",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Action_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "cure",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }


    [Fact]
    public async Task Achievement_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "For the hoard",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task BNpcname_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "bhoot",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ClassJob_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute("summoner", Language.English);

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Companion_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "lesser panda",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ContentFinderCondition_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "sastasha",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task CraftAction_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Groundwork",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Emote_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "pet",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task ENpcResident_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "G'raha Tia",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task EventItemLite_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Ulan's note",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Fate_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Drink Me",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Leve_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Practical Command",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Mount_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Demi-Ozma",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Orchestrion_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "pa-paya",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task PlaceName_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Limsa Lominsa",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Quest_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "For Want Of a Memory",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Status_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Vulnerability Up",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Title_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Goddess of Magic",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Trait_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Maim and Mend",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }

    [Fact]
    public async Task Weather_SearchContentByName_ReturnsMatchingResults()
    {
        IEnumerable<Content> results = await _searchContentByNameUseCase.Execute(
            "Fair Skies",
            Language.English
        );

        Snapshot.Match(results.ToList());
    }
}
