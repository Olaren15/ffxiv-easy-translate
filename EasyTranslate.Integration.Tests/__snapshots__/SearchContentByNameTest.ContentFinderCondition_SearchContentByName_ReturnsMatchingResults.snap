[
  {
    "Type": "Instance",
    "IconId": 61801,
    "EnglishName": "Sastasha",
    "FrenchName": "Sastasha",
    "GermanName": "Sastasha",
    "JapaneseName": "天然要害 サスタシャ浸食洞"
  },
  {
    "Type": "Place",
    "IconId": null,
    "EnglishName": "Sastasha",
    "FrenchName": "Sastasha",
    "GermanName": "Sastasha-Höhle",
    "JapaneseName": "サスタシャ浸食洞"
  },
  {
    "Type": "Place",
    "IconId": null,
    "EnglishName": "Sastasha",
    "FrenchName": "Sastasha",
    "GermanName": "Sastasha",
    "JapaneseName": "天然要害 サスタシャ浸食洞"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "Sastasha pugil",
    "FrenchName": "pugil de Sastasha",
    "GermanName": "Sastasha-Pugil",
    "JapaneseName": "サスタシャ・プギル"
  },
  {
    "Type": "Npc",
    "IconId": null,
    "EnglishName": "Sastasha orobon",
    "FrenchName": "orobon de Sastasha",
    "GermanName": "Sastasha-Orobon",
    "JapaneseName": "サスタシャ・オロボン"
  },
  {
    "Type": "Instance",
    "IconId": 61801,
    "EnglishName": "Sastasha (Hard)",
    "FrenchName": "Sastasha (brutal)",
    "GermanName": "Sastasha (schwer)",
    "JapaneseName": "逆襲要害 サスタシャ浸食洞 (Hard)"
  },
  {
    "Type": "Place",
    "IconId": null,
    "EnglishName": "Sastasha (Hard)",
    "FrenchName": "Sastasha (brutal)",
    "GermanName": "Sastasha-Höhle (schwer)",
    "JapaneseName": "サスタシャ浸食洞 (Hard)"
  },
  {
    "Type": "Place",
    "IconId": null,
    "EnglishName": "Sastasha Seagrot",
    "FrenchName": "Grotte marine de Sastasha",
    "GermanName": "Sastasha-Höhle",
    "JapaneseName": "浸食洞"
  },
  {
    "Type": "Achievement",
    "IconId": 1001,
    "EnglishName": "Mapping the Realm: Sastasha",
    "FrenchName": "Cartographe du continent : Sastasha",
    "GermanName": "Kartograph: Sastasha",
    "JapaneseName": "サスタシャ浸食洞を踏破せし者"
  },
  {
    "Type": "Achievement",
    "IconId": 1001,
    "EnglishName": "Remapping the Realm: Sastasha",
    "FrenchName": "Cartographe du continent : Sastasha (brutal)",
    "GermanName": "Kartograph: Sastasha-Höhle (schwer)",
    "JapaneseName": "サスタシャ浸食洞を再び踏破せし者"
  }
]
